pydantic/typing.py

```diff
@@ -91,6 +94,11 @@ def get_all_type_hints(obj: Any, globalns: Any = None, localns: Any = None) -> A
 AnnotatedTypeNames = {'AnnotatedMeta', '_AnnotatedAlias'}
 
 
+LITERAL_TYPES: Set[Union[TypingSpecialForm, TypingExtensionSpecialForm]] = {Literal}
```

Comment:
My advice would be to go for a slightly less precise annotation here -- we definitely regard `_SpecialForm` as an undocumented implementation detail, both at CPython and at `typing_extensions` :)

```suggestion
LITERAL_TYPES: Set[object] = {Literal}
```

Reply:
Agree, Changed!